adapters/amqp-vertx/src/main/java/org/eclipse/hono/adapter/amqp/AmqpContext.java

```diff
@@ -87,15 +95,19 @@ ResourceIdentifier getResourceIdentifier() {
     }
 
     /**
-     * Sets an AMQP 1.0 message delivery state to rejected. The error description for rejecting the delivery is
-     * retrieved from the provided exception parameter.
+     * Sets an AMQP 1.0 message delivery state to either RELEASED in the case of a <em>ServerErrorException</em> or REJECTED in the
+     * case of a <em>ClientErrorException</em>. In the REJECTED case, the supplied exception will provide
+     * the error condition value and description as reason for rejection.
      *
-     * @param condition The error condition value to include in the Rejected disposition frame.
-     * @param error The client exception providing a reason for rejecting the message.
+     * @param error The service invocation exception.
      */
-    void reject(final Symbol condition, final ClientErrorException error) {
-        final ErrorCondition errorCondition = ProtonHelper.condition(condition, error.getMessage());
-        MessageHelper.rejected(delivery, errorCondition);
+    void handleFailure(final ServiceInvocationException error) {
```

Comment:
what if error is `null`

Reply:
`handleFailure` expects that the throwable is not `null`. so i can add a null-check at the beginning and modify the JavaDoc. 
question: how can it ever be `null`?

Reply:
> question: how can it ever be null?

well, we cannot prevent anyone from creating a new class in the same package and invoke the method passing in `null`, can we? Just because *we* are not doing it, doesn't mean that anybody else isn't doing it. Documenting the behavior in case someone does is a safe way of encouraging people to actually care, because they know what to expect from the method.

Reply:
:+1: